openshift/machine-config-operator
# Allow blocking payload reg if it has mirrors configured

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

This relates to https://issues.redhat.com/browse/RFE-2166



**- What I did**
Some disconnected use cases require adding the payload reg
to the list of blocked registries for certain compliance reasons.
So allow adding the payload registry to the list of blocked registries
if we find a mirror configuration set for it.

**- How to verify it**
Bring up a cluster and check what registry the payload is being pulled from. Then test the following cases:
- Add the payload registry to the blocked registry list --> the payload reg should not be added under blocked registries in /etc/containers/registries.conf
- Create a mirror rule using ICSP for the payload reg and then add the payload reg to the blocked registry list --> the payload reg should be added under the blocked registries in /etc/containers/registries.conf and there should be a warning in the logs about adding the payload registry to the list of blocked registries.

**- Description for the changelog**
Allow blocking the payload registry if there are mirror rules set for it.
